### PR TITLE
Map authorities from access token claim

### LINKED ISSUE #9735: Secured endpoints do not work with OIDC
##### **Overview of the issue**

We systematically get a 401 when using `@Secured(AuthoritiesConstants.ADMIN)` with OIDC configuration.
The authorityMapper is not called, and also it's not accurate when compared to the JWT token.
What's the best way for fixing the issue?

##### **Motivation for or Use Case**

Securing endpoint

##### **Reproduce the error**

add a @Secured annotation on a resource endpoint (on a microservice on my side).
Try to call it, and dejoy... (tested on a microservice).
Also, I noticed that there were some agressive cast of Authentication to OAuth2AuthenticationToken, which is not always the case (for exemple service to service communication with Feign). this is not particularly related but to take into account while fixing. 

##### **Suggest a Fix**

Don't know, an AuthenticationManager implementation + a good implementation authorityMapper. Or maybe a token enhancer?

##### **JHipster Version(s)**

6.0.0

##### **JHipster configuration**

```
application {
  config {
    baseName pmi
    packageName com.corp.ecwp.pmi
    applicationType gateway
    authenticationType oauth2
    cacheProvider hazelcast
    clientFramework angularX
    clientPackageManager yarn
    enableHibernateCache true
    languages [en, fr]
    searchEngine elasticsearch
    serverPort 8080
    serviceDiscoveryType eureka
    testFrameworks [protractor]
    useSass true
  }
  entities X
}

application {
  config {
    baseName marketplace
    applicationType microservice
    authenticationType oauth2
    cacheProvider hazelcast
    clientPackageManager yarn
    databaseType sql
    enableHibernateCache true
    languages [en, fr]
    packageName com.corp.service.marketplace
    searchEngine elasticsearch
    serverPort 8103
    serviceDiscoveryType eureka
    useSass true
  }
  entities Y
}
```

-   [x] Checking this box is mandatory (this is just to show you read everything)

### PR DESCRIPTION
**Pull request by vishal423**
Closes #9735 

-   Please make sure the below checklist is followed for Pull Requests.

-   [ ] [Travis tests](https://travis-ci.org/jhipster/generator-jhipster/pull_requests) are green
-   [x] Tests are added where necessary
-   [x] Documentation is added/updated where necessary
-   [x] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/master/CONTRIBUTING.md) are followed

### CODE REVIEW

**Comment by vishal423** [comment]
@mraible, ping.

File: generators/server/templates/src/main/java/package/config/SecurityConfiguration.java.ejs [hunk 1]
```diff
@@ -285,34 +289,57 @@ public class SecurityConfiguration extends WebSecurityConfigurerAdapter {
...
+                    getRolesFromClaims(jwt.getClaims())
+                );
+            }
+        };
+    }
+
+    @SuppressWarnings("unchecked")
+    private Collection<String> getRolesFromClaims(Map<String, Object> claims) {
+        return (Collection<String>) claims.getOrDefault(
+            "groups",
```
Comment (mraible) [inline generators/server/templates/src/main/java/package/config/SecurityConfiguration.java.ejs]: Minor request: move this up one line.

File: generators/server/templates/src/main/java/package/config/SecurityConfiguration.java.ejs [hunk 2]
```diff
@@ -285,34 +289,57 @@ public class SecurityConfiguration extends WebSecurityConfigurerAdapter {
...
+            claims.getOrDefault(
+                "roles",
+                new ArrayList<>()
```
Comment (mraible) [inline generators/server/templates/src/main/java/package/config/SecurityConfiguration.java.ejs]: Collapse the above three lines into one.

**Comment by pascalgrimaud** [comment]
thanks @vishal423 !

**Pull request merged**